http/src/client/builder.rs

```diff
@@ -65,32 +59,41 @@ impl ClientBuilder {
         self
     }
 
-    /// Sets the proxy to use for all HTTP requests.
+    /// Set a pre-configured Hyper client builder to build off of.
     ///
-    /// This accepts a `reqwest::Proxy`.
-    pub fn proxy(mut self, proxy: Proxy) -> Self {
-        self.proxy.replace(proxy);
-
-        self
-    }
-
-    /// Set whether to proxy over HTTP.
+    /// The proxy and timeout settings in the Hyper client will be overridden by
+    /// those in this builder.
     ///
-    /// The default is `false`.
-    pub fn proxy_http(mut self, proxy_http: bool) -> Self {
-        self.proxy_http = proxy_http;
+    /// The default client uses Rustls as its TLS backend.
+    pub fn hyper_client(mut self, client: HyperClient<HttpsConnector<HttpConnector>>) -> Self {
+        self.hyper_client.replace(client);
 
         self
     }
 
-    /// Set a pre-configured reqwest client builder to build off of.
+    /// Set the proxy to use for all HTTP(S) requests.
     ///
-    /// The proxy and timeout settings in the reqwest client will be overridden by
-    /// those in this builder.
+    /// **Note** that this isn't currently a traditional proxy, but is for
+    /// working with something like [twilight's HTTP proxy server].
     ///
-    /// The default client uses Rustls as its TLS backend.
-    pub fn reqwest_client(mut self, client: ReqwestClientBuilder) -> Self {
-        self.reqwest_client.replace(client);
+    /// # Examples
+    ///
+    /// Set the proxy to `twilight_http_proxy.internal`:
+    ///
+    /// ```rust
+    /// use twilight_http::Client;
+    ///
+    /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+    /// let client = Client::builder()
+    ///     .proxy("twilight_http_proxy.internal", true)
+    ///     .build();
+    /// # Ok(()) }
+    /// ```
+    ///
+    /// [twilight's HTTP proxy server]: https://github.com/twilight-rs/http-proxy
+    pub fn proxy(mut self, proxy_url: impl Into<Box<str>>, use_http: bool) -> Self {
```

Comment:
Can be convinced to not care, but why not just take `Into<String>` and call `.into().into_boxed_str()` internally instead of the arguably bad function signature?

Reply:
As far as I know that has barely any cost. Resolved